main.py

```diff
@@ -971,6 +1053,72 @@ def suggest(limit=10):
     return better_jsonify(suggestions=suggestions)
 
 
+@app.route("/locations", methods=["GET"])
+@max_age(seconds=60 * 5)
+def locations():
+    """ Render locations page """
+    kind = request.args.get("kind")
+    period = request.args.get("period")
+
```

Comment:
Probably need more error checking on `kind` and `period` arguments here, since they are effectively user-supplied